### PR TITLE
chore: remove Node.js version check

### PR DESCRIPTION
**Pull request by tim-sh**
fixes https://github.com/cap-js/cds-intellij/issues/47

On startup with a Node version mismatch, this will show the original error from the cds-lsp server:
grafik

### CODE REVIEW

File: src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java [hunk 1]
```diff
@@ -9,86 +9,75 @@
...
         OSProcessHandler handler = new OSProcessHandler(getCommandLine()) {
             @Override
             protected BaseOutputReader.@NotNull Options readerOptions() {
                 return BaseOutputReader.Options.forMostlySilentProcess();
             }
+            @Override
+            protected void onOSProcessTerminated(int exitCode) {
+                super.onOSProcessTerminated(exitCode);
+                if (started && exitCode != 0) {
+                    showUserError(getProcess());
```
Comment (manwithafox) [inline src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java:48]: Is this the UI thread? If yes, why does `started` to be `volatile`? If no, is `UserError.show(...)` synchronizing into the UI therad, then?

**Review by manwithafox — COMMENTED** [review]
Just some comments, mainly about Java multithreading. 
I see three cases that should have been tested (probably manually):

1) node does not exist
2) node exists but is too old
3) node version is appropriate

File: src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java [hunk 2]
```diff
@@ -9,86 +9,75 @@
...
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class ServerDescriptor extends ProjectWideLspServerDescriptor {
 
     private static final String RELATIVE_SERVER_PATH = "cds-lsp/node_modules/@sap/cds-lsp/dist/main.js";
-    private static final String RELATIVE_SERVER_PKG_PATH = "cds-lsp/node_modules/@sap/cds-lsp/package.json";
     private static final String RELATIVE_MITM_PATH = "cds-lsp/mitm.js";
     private static final String RELATIVE_LOG_PATH = "cds-lsp/stdio.json";
 
+    private volatile Boolean started = false;
```
[the remark this answers is not in the document]
Comment (tim-sh) [reply src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java:30]: Changed it to the primitive type for efficiency, although both is fine with the overriden method.

File: src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java (hunk 1 above)
Comment (tim-sh) [reply src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java:48]: I just verified in the decompiled sources and at runtime that it's all the same thread, so your comment is perfectly valid. Removing `volatile`.

File: src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java [hunk 3]
```diff
@@ -9,86 +9,75 @@
...
-        }
-        if (versionStr.startsWith("v")) {
-            versionStr = versionStr.substring(1);
-        }
-
-        ComparableVersion requiredVersion = new ComparableVersion(requiredVersionStr);
-        ComparableVersion version = new ComparableVersion(versionStr);
-        if (version.compareTo(requiredVersion) < 0) {
-            UserError.show("CDS Language Server requires Node.js version %s, but found earlier Node.js version %s. Please update your Node.js installation.".formatted(requiredVersionStr, versionStr));
+            process.exitValue();
```
[the remark this answers is not in the document]
Comment (tim-sh) [reply src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java]: Of course. I don't exactly remember why this indirection, but there you go. :) Thanks for your hints

**Comment by tim-sh** [comment]
> Just some comments, mainly about Java multithreading. I see three cases that should have been tested (probably manually):
> 
>     1. node does not exist

… or not readable/executable. In such cases, the Language Services widget (status bar) get a red exclamation mark:
grafik
Clicking on it and the the other exclamation mark opens stderr output, which begins with:
> com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "node": error=13, Keine Berechtigung

So I think for the time being that's clear enough and we'll improve here once we/the user selects from available Node runtimes.

>     2. node exists but is too old
>     3. node version is appropriate

both tested fine

**Review by manwithafox — APPROVED** [review]
LGTM

**Pull request merged**